fulltext/backends/__doc.py

```diff
@@ -16,6 +18,8 @@ class Backend(BaseBackend):
     def check(self):
         assert_cmd_exists('antiword')
         assert_cmd_exists('abiword')
+        if "FULLTEXT_TESTING" in os.environ:
+            assert_cmd_exists('pdfinfo')
```

Comment:
I think I would prefer a boolean flag argument to `check()`, to check for title extraction requirements too.

Reply:
I'm not sure I understand. =)

Reply:
Sorry, I could have explained better. Also, I am making an assumption.

Assumption: I assume this `FULLTEXT_TESTING` environment variable is being set while unit testing, and exists because unit tests test title extraction and therefore require additional CLI tools. You want `check()` to look for these additional tools during unit test, so that failure of these tests is early and clear.

So now my suggestion: I think that `check()` should accept a flag, such as `check(required_title=False)` and that the flag should be used in place of the environment variable. Then the unit test can call `check()`, provide a `True` which is sent to all the backend `check()` functions, achieving the same result. Then this flag becomes part of the fulltext API, and not some test-time configuration variable.

Reply:
If you want it to become part of the API then I suppose it means adding a new arg to `get_with_title()` function. But is it really worth it? I.e. right now there is no such API for `get`: when a dep is not installed it just falls back on using bin backend.

Maybe we can add a generic `require_deps=False` to both functions and unit tests will always have it set to `True`. Makes sense?

Reply:
docstring for both `get` and `get_with_title`:

```
require_deps: when False, if a certain dep is not installed for a certain backend/extension the "bin" backend will be used instead and emit a warning. When True it will fail immediately.
```

Reply:
I think the flag is useful for `check()` alone.